lib/rubocop/cop/lint/missing_cop_enable_directive.rb

```diff
@@ -22,21 +25,53 @@ module Lint
       #   x= 0
       #   # EOF
       #
+      # @example
+      #   Lint/MissingCopEnableDirective
+      #     MaximumRangeSize: 2
+      #
+      #   # good
+      #   # rubocop:disable Layout/SpaceAroundOperators
+      #   x= 0
+      #   # With the previous, there are 2 lines on which cop is disabled.
+      #   # rubocop:enable Layout/SpaceAroundOperators
+      #
+      #   # bad
+      #   # rubocop:disable Layout/SpaceAroundOperators
+      #   x= 0
+      #   x += 1
+      #   # Including this, that's 3 lines on which the cop is disabled.
+      #   # rubocop:enable Layout/SpaceAroundOperators
+      #
       class MissingCopEnableDirective < Cop
         MSG = 'Re-enable %s cop with `# rubocop:enable` after disabling it.'
               .freeze
 
         def investigate(processed_source)
+          max_range = cop_config['MaximumRangeSize']
           processed_source.disabled_line_ranges.each do |cop, line_ranges|
             line_ranges.each do |line_range|
-              next unless line_range.max == Float::INFINITY
+              # This has to remain a strict inequality to handle
+              # the case when max_range is Float::INFINITY
+              next if line_range.max - line_range.min < max_range + 2
               range = source_range(processed_source.buffer,
                                    line_range.min,
                                    (0..0))
-              add_offense(range, location: range, message: format(MSG, cop))
+              add_offense(range,
+                          location: range,
+                          message: format(message(max_range), cop))
             end
           end
         end
+
+        private
+
+        def message(max_range)
+          if max_range == Float::INFINITY
```

Comment:
Perhaps `max_range.infinite?` is clearer?

Reply:
You gave me idea about a new cop. 😉